src/libslic3r/PerimeterGenerator.cpp

```diff
@@ -2436,6 +2436,240 @@ void PerimeterGenerator::process_no_bridge(Surfaces& all_surfaces, coord_t perim
     }
 }
 
+// ORCA:
+// Inner Outer Inner wall ordering mode perimeter order optimisation functions
+
+/**
+ * @brief Converts the junctions of an ExtrusionLine to a vector of Points.
+ *
+ * @param line The input ExtrusionLine containing junctions to be converted.
+ * @return std::vector<Point> A vector containing points extracted from the junctions of the input ExtrusionLine.
+ */
+std::vector<Point> extrusionLineToPointsIOI(const Arachne::ExtrusionLine& line) {
+    std::vector<Point> points;
+    for (const auto& junction : line.junctions) {
+        points.push_back(junction.p);
+    }
+    return points;
+}
+
+/**
+ * @brief Calculates the squared distance between two points.
+ *
+ * @param a The first point.
+ * @param b The second point.
+ * @return double The squared distance between the two points.
+ */
+double squaredDistance(const Point& a, const Point& b) {
+    return (a.x() - b.x()) * (a.x() - b.x()) + (a.y() - b.y()) * (a.y() - b.y());
+}
+
+/**
+ * @brief Calculates the squared distance between a point and a line segment.
+ *
+ * @param p The point.
+ * @param v The starting point of the line segment.
+ * @param w The ending point of the line segment.
+ * @return double The squared distance between the point and the line segment.
+ */
+double squaredDistanceToSegment(const Point& p, const Point& v, const Point& w) {
```

Comment:
Looks like this is same as MultiPoint::distance_to function.
I suggest we use that one.
If they are different, we should make it a member function of MultiPoint class.

Reply:
Unfortunately they are not functionally equivalent. The distance_to function looks at the Euclidean distance between a given point and the closest point to it. The squaredDistanceToLineSegment  calculates the squared distance from a point to a line segment defined by two points. 

While I could just simplify this down to simply the distance between point A in line Y and point B in line X I feel that this wouldn't be accurate enough as it doesnt check the distance between the actual line segments but rather the points that define them. For example a curve defined by say 10 points and a line defined by 2 points. If I check distance between points for those it would yield the wrong physical distance that they would have when printed, as the line defined by the 2 points has them at each end vs. the curve having them dotted around throughout the curve. 

Hope that makes some sense? Sorry I'm struggling to articulate it without drawing a picture :D 

This function has now moved to the MultiPoint class.